packages/astro/src/core/app/index.ts

```diff
@@ -201,8 +213,46 @@ export class App {
 				route: routeData,
 				status,
 			});
+			const paramsAndPropsResp = await getParamsAndProps({
+				mod: mod as any,
+				route: renderContext.route,
+				routeCache: this.#env.routeCache,
+				pathname: renderContext.pathname,
+				logging: this.#env.logging,
+				ssr: this.#env.ssr,
+			});
+
+			if (paramsAndPropsResp === GetParamsAndPropsError.NoMatchingStaticPath) {
```

Comment:
We don't want this error because static paths are a SSG only thing. `getStaticPaths` is not called in SSR. 

Actually I'm not sure why you added `getParamsAndProps` call in the first place. Is it needed for some reason? How were params being populated before this?

Reply:
This is just code that I copied from somewhere else to create `APIContext`. I am happy to remove the error if you think it's unnecessary.

Reply:
> Actually I'm not sure why you added getParamsAndProps call in the first place. Is it needed for some reason? How were params being populated before this?

In order to create the APIContext, I needed props and params. Should I just pass empty objects?

Reply:
No, we do need those. I should want to make sure we're not calling this code more than once. Where is `params` being populated normally, before this change? I guess you are moving this code up the tree and removing it from the previous place?

The error is needed in SSG mode but not in SSR. This module is SSR-only (currently).